tictactoe/adapter-api/src/main/kotlin/com/anaplan/engineering/azuki/tictactoe/adapter/api/TicTacToeActionGeneratorFactory.kt

```diff
@@ -0,0 +1,25 @@
+package com.anaplan.engineering.azuki.tictactoe.adapter.api
+
+import com.anaplan.engineering.azuki.core.system.ActionGenerator
+import com.anaplan.engineering.azuki.core.system.ActionGeneratorFactory
+
+interface TicTacToeActionGeneratorFactory : ActionGeneratorFactory {
+
+    /**
+     * Generates a valid play order and assigns it to the given name.
+     * A play order must not already exist with this name.
```

Comment:
❓ Is this too restrictive, or should the action generator just e.g. silently ignore cases where it's asked to generate a play order twice?

Reply:
No, I think that's fine. We've got the same ID uniqueness requirement all over the place. In normal operation, these will be automatically generated, anyway. If a user is hand-crafting a guided scenario, then it's their responsibility to ensure this requirement is met. Silently ignoring the duplicate names would just introduce subtle bugs and lead to confusion.